libyul/optimiser/ReasoningBasedSimplifier.cpp

```diff
@@ -56,14 +56,7 @@ std::optional<string> ReasoningBasedSimplifier::invalidInCurrentEnvironment()
 
 void ReasoningBasedSimplifier::operator()(VariableDeclaration& _varDecl)
 {
-	if (_varDecl.variables.size() != 1 || !_varDecl.value)
```

Comment:
Why is this change not just part of the first commit?

Reply:
I would say either combine this commit with the first or also put the creation of `Solver::encodeVariableDeclaration` into this (second) commit.